apps/api/src/app/tenant/tenant.controller.ts

```diff
@@ -0,0 +1,178 @@
+import { PaginatedResponseDto } from '../shared/dtos/pagination-response';
+import { GetTenantsRequestDto } from './dtos/get-tenants.request.dto';
+import { GetTenants } from './usecases/get-tenants/get-tenants.usecase';
+import { GetTenantsCommand } from './usecases/get-tenants/get-tenants.command';
+
+@Controller('/tenants')
+@ApiTags('Tenants')
+@UseInterceptors(ClassSerializerInterceptor)
+@UseGuards(JwtAuthGuard)
+@ApiExcludeController()
+export class TenantController {
+  constructor(
+    private createTenantUsecase: CreateTenant,
+    private updateTenantUsecase: UpdateTenant,
+    private getTenantUsecase: GetTenant,
+    private deleteTenantUsecase: DeleteTenant,
+    private getTenantsUsecase: GetTenants
+  ) {}
+
+  @Get('')
+  @ExternalApiAccessible()
+  @UseGuards(JwtAuthGuard)
+  @ApiOkPaginatedResponse(GetTenantResponseDto)
+  @ApiOperation({
+    summary: 'Get tenants',
+    description: 'Returns a list of tenants, could paginated using the `page` and `limit` query parameter',
+  })
+  async getTenants(
+    @UserSession() user: IJwtPayload,
+    @Query() query: GetTenantsRequestDto
+  ): Promise<PaginatedResponseDto<GetTenantResponseDto>> {
+    return await this.getTenantsUsecase.execute(
+      GetTenantsCommand.create({
+        organizationId: user.organizationId,
+        environmentId: user.environmentId,
+        page: query.page ? Number(query.page) : 0,
+        limit: query.limit ? Number(query.limit) : 10,
+      })
+    );
+  }
+
+  @Get('/:identifier')
+  @ApiResponse(GetTenantResponseDto)
+  @ApiOperation({
+    summary: 'Get tenant',
+    description: `Get tenant by your internal id used to identify the tenant`,
+  })
+  @ExternalApiAccessible()
+  getTenantById(
+    @UserSession() user: IJwtPayload,
+    @Param('identifier') identifier: string
+  ): Promise<GetTenantResponseDto> {
+    return this.getTenantUsecase.execute(
+      GetTenantCommand.create({
+        environmentId: user.environmentId,
+        organizationId: user.organizationId,
+        identifier: identifier,
+      })
+    );
+  }
+
+  @Post('/')
+  @ExternalApiAccessible()
+  @ApiResponse(CreateTenantResponseDto)
+  @ApiOperation({
+    summary: 'Create tenant',
+    description: 'Create tenant under the current environment',
+  })
+  @ApiConflictResponse({
+    description: 'A tenant with the same identifier is already exist.',
+  })
+  async createTenant(
+    @UserSession() user: IJwtPayload,
+    @Body() body: CreateTenantRequestDto
+  ): Promise<CreateTenantResponseDto> {
+    return await this.createTenantUsecase.execute(
+      CreateTenantCommand.create({
+        environmentId: user.environmentId,
+        organizationId: user.organizationId,
+        identifier: body.identifier,
+        name: body.name,
+        data: body.data,
+      })
+    );
+  }
+
+  @Put('/:identifier')
+  @ExternalApiAccessible()
+  @ApiResponse(UpdateTenantResponseDto)
```

Comment:
Missing the definition for `ApiNotFound`.

Reply:
added.
i wonder if we can make those more auto-generated or make it mandatory on the compilation time.